File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/composite/Directory.java
package patterns.composite;

import java.util.*;

public class Directory extends Node {
    public Directory(String name) {
        super(name);
        contents = new ArrayList<Node>();
    }
    public void add(Node n) {
        contents.add(n);
    }
    public Node[] contents() {
        Node[] stuff = new Node[contents.size()];
        return contents.toArray(stuff);
    }
    public long length() {
        long total = 0;
        Node[] content = contents();
        for(int i=0;i<content.length;i++) {
            total += content[i].length();
        }
        return total;
    }
    protected void print(int depth) {
        printPrefix(depth);
        depth++;
        System.out.println(toString() + ":");
        Node[] content = contents();
        for(int i=0;i<content.length;i++) {
            content[i].print(depth);
        }
    }
    private List<Node> contents;
}

File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Composite/Directory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Composite {
    internal class Directory : Node {
        public Directory(String name) : base(name) {
            contents = new List<Node>();
        }

        public void Add(Node n) {
            contents.Add(n);
        }

        public Node[] Contents() {
            return contents.ToArray();
        }

        public override long Length {
            get { return contents.Sum((n) => n.Length); }
        }

        public override void Print(int depth) {
            PrintPrefix(depth);
            depth++;
            Console.WriteLine(ToString() + ":");
            contents.ForEach((n) => n.Print(depth));
        }

        private readonly List<Node> contents;
    }
}